ocsp_status.go

```diff
@@ -38,35 +39,58 @@ func checkFromFile(respFile string) error {
 	return nil
 }
 
+// checkFromCert takes a path to an ASN.1 DER encoded certificate file and constructs and sends an OCSP request
+// then parses and lints the OCSP response
 func checkFromCert(certFile string, isPost bool, ocspURL string, dir string, hash crypto.Hash) error {
 	reqMethod := http.MethodGet
 	if isPost {
 		reqMethod = http.MethodPost
 	}
 
-	ocspReq, issuerCert, err := ocsptools.CreateOCSPReqFromCert(certFile, ocspURL, reqMethod, hash)
+	leafCert, err := ocsptools.ParseCertificateFile(certFile)
 	if err != nil {
-		return fmt.Errorf("Error creating OCSP Request from certificate file: %w", err)
+		return fmt.Errorf("Error parsing certificate from certificate file: %w", err)
+	}
+
+	issuerCert, err := ocsptools.GetIssuerCertFromLeafCert(leafCert)
+	if err != nil {
+		return fmt.Errorf("Error getting issuer certificate from certificate: %w", err)
+	}
+
+	ocspReq, err := ocsptools.CreateOCSPReq(ocspURL, leafCert, issuerCert, reqMethod, hash)
+	if err != nil {
+		return fmt.Errorf("Error creating OCSP Request: %w", err)
 	}
 
 	ocspResp, err := ocsptools.GetOCSPResponse(ocspReq)
 	if err != nil {
-		return fmt.Errorf("Error getting OCSP Response from certificate file: %w", err)
+		return fmt.Errorf("Error getting OCSP Response: %w", err)
 	}
 
 	return ocsptools.ParseAndLint(ocspResp, issuerCert)
-
 }
 
-func checkFromURL(serverURL string, shouldPrint bool, isPost bool, ocspURL string, dir string, hash crypto.Hash) error {
+// checkFromURL takes a server URL and constructs and sends an OCSP request to check that URL's certificate
+// then parses and lints the OCSP response
+func checkFromURL(serverURL string, shouldPrint bool, isPost bool, noStaple bool, ocspURL string, dir string, hash crypto.Hash) error {
 	tlsConn, err := createConn(serverURL)
 	if err != nil {
 		return err
 	}
 
 	certChain := tlsConn.ConnectionState().PeerCertificates
-	leafCert := certChain[0]
-	issuerCert := certChain[1]
+
+	if len(certChain) == 0 {
+		return fmt.Errorf("No certificate present for %s", serverURL)
+	}
+
+	// Is this right?
+	if len(certChain) == 1 {
+		return fmt.Errorf("Certificate for %s is a root certificate", serverURL)
+	}
```

Comment:
nit: maybe turn into if, else if, else?

Reply:
I'm going to omit the `else` because I think the code will just be harder to read with another level of indentation

Reply:
fair, I just realized you could just make len(certChain) a variable to avoid calling len() twice, just that's a nit.